Flux end-to-end tests appear to be flaky again

I saw this fail in CI, with no output produced. The relevant log is below.

This test is pretty simple and just basically calls `map`, which makes me concerned that we have a very infrequent flakiness in our test framework.
```
    --- FAIL: TestFluxEndToEnd/to_time (0.25s)
        end_to_end_test.go:256: diff table was not empty
        end_to_end_test.go:249: Result: diff
        end_to_end_test.go:249: Table: keys: [_start, _stop, _field, _measurement]
        end_to_end_test.go:249:                    _start:time                      _stop:time           _field:string     _measurement:string            _diff:string                      _time:time                     _value:time  
        end_to_end_test.go:249: ------------------------------  ------------------------------  ----------------------  ----------------------  ----------------------  ------------------------------  ------------------------------  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k0                       m                       -  2018-05-22T19:53:00.000000000Z  1970-01-01T00:00:00.000000000Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k0                       m                       -  2018-05-22T19:53:01.000000000Z  2018-05-22T19:53:26.033000000Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k0                       m                       -  2018-05-22T19:53:02.000000000Z  2018-05-22T19:53:26.033066000Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k0                       m                       -  2018-05-22T19:53:03.000000000Z  2018-05-22T18:00:00.000000000Z  
        end_to_end_test.go:249: Table: keys: [_start, _stop, _field, _measurement]
        end_to_end_test.go:249:                    _start:time                      _stop:time           _field:string     _measurement:string            _diff:string                      _time:time                     _value:time  
        end_to_end_test.go:249: ------------------------------  ------------------------------  ----------------------  ----------------------  ----------------------  ------------------------------  ------------------------------  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k1                       m                       -  2018-05-22T19:53:10.000000000Z  2018-05-22T19:53:26.000000000Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k1                       m                       -  2018-05-22T19:53:11.000000000Z  2018-05-22T19:53:26.033000000Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k1                       m                       -  2018-05-22T19:53:12.000000000Z  2018-05-22T19:53:26.033066000Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k1                       m                       -  2018-05-22T19:53:13.000000000Z  2018-05-22T19:00:00.000000000Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k1                       m                       -  2018-05-22T19:53:14.000000000Z  2018-05-22T19:00:00.000000000Z  
        end_to_end_test.go:249: Table: keys: [_start, _stop, _field, _measurement]
        end_to_end_test.go:249:                    _start:time                      _stop:time           _field:string     _measurement:string            _diff:string                      _time:time                     _value:time  
        end_to_end_test.go:249: ------------------------------  ------------------------------  ----------------------  ----------------------  ----------------------  ------------------------------  ------------------------------  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k2                       m                       -  2018-05-22T19:53:20.000000000Z  1970-01-01T00:00:00.000000000Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k2                       m                       -  2018-05-22T19:53:21.000000000Z  2015-12-31T23:59:59.999999999Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k2                       m                       -  2018-05-22T19:53:22.000000000Z  2018-05-22T19:53:26.033000000Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k2                       m                       -  2018-05-22T19:53:23.000000000Z  2018-05-22T19:00:00.000000000Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k2                       m                       -  2018-05-22T19:53:24.000000000Z  2018-05-22T21:00:00.000000000Z  
        end_to_end_test.go:249: Table: keys: [_start, _stop, _field, _measurement]
        end_to_end_test.go:249:                    _start:time                      _stop:time           _field:string     _measurement:string            _diff:string                      _time:time                     _value:time  
        end_to_end_test.go:249: ------------------------------  ------------------------------  ----------------------  ----------------------  ----------------------  ------------------------------  ------------------------------  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k3                       m                       -  2018-05-22T19:53:30.000000000Z  1970-01-01T00:00:00.000000000Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k3                       m                       -  2018-05-22T19:53:31.000000000Z  1969-12-31T23:59:59.999999999Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k3                       m                       -  2018-05-22T19:53:32.000000000Z  2018-05-22T19:53:26.033066000Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k3                       m                       -  2018-05-22T19:53:33.000000000Z  2018-05-22T18:00:00.000000000Z  
        end_to_end_test.go:249: Result: want
        end_to_end_test.go:249: Table: keys: [_start, _stop, _field, _measurement]
        end_to_end_test.go:249:                    _start:time                      _stop:time           _field:string     _measurement:string                      _time:time                     _value:time  
        end_to_end_test.go:249: ------------------------------  ------------------------------  ----------------------  ----------------------  ------------------------------  ------------------------------  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k0                       m  2018-05-22T19:53:00.000000000Z  1970-01-01T00:00:00.000000000Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k0                       m  2018-05-22T19:53:01.000000000Z  2018-05-22T19:53:26.033000000Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k0                       m  2018-05-22T19:53:02.000000000Z  2018-05-22T19:53:26.033066000Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k0                       m  2018-05-22T19:53:03.000000000Z  2018-05-22T18:00:00.000000000Z  
        end_to_end_test.go:249: Table: keys: [_start, _stop, _field, _measurement]
        end_to_end_test.go:249:                    _start:time                      _stop:time           _field:string     _measurement:string                      _time:time                     _value:time  
        end_to_end_test.go:249: ------------------------------  ------------------------------  ----------------------  ----------------------  ------------------------------  ------------------------------  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k1                       m  2018-05-22T19:53:10.000000000Z  2018-05-22T19:53:26.000000000Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k1                       m  2018-05-22T19:53:11.000000000Z  2018-05-22T19:53:26.033000000Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k1                       m  2018-05-22T19:53:12.000000000Z  2018-05-22T19:53:26.033066000Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k1                       m  2018-05-22T19:53:13.000000000Z  2018-05-22T19:00:00.000000000Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k1                       m  2018-05-22T19:53:14.000000000Z  2018-05-22T19:00:00.000000000Z  
        end_to_end_test.go:249: Table: keys: [_start, _stop, _field, _measurement]
        end_to_end_test.go:249:                    _start:time                      _stop:time           _field:string     _measurement:string                      _time:time                     _value:time  
        end_to_end_test.go:249: ------------------------------  ------------------------------  ----------------------  ----------------------  ------------------------------  ------------------------------  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k2                       m  2018-05-22T19:53:20.000000000Z  1970-01-01T00:00:00.000000000Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k2                       m  2018-05-22T19:53:21.000000000Z  2015-12-31T23:59:59.999999999Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k2                       m  2018-05-22T19:53:22.000000000Z  2018-05-22T19:53:26.033000000Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k2                       m  2018-05-22T19:53:23.000000000Z  2018-05-22T19:00:00.000000000Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k2                       m  2018-05-22T19:53:24.000000000Z  2018-05-22T21:00:00.000000000Z  
        end_to_end_test.go:249: Table: keys: [_start, _stop, _field, _measurement]
        end_to_end_test.go:249:                    _start:time                      _stop:time           _field:string     _measurement:string                      _time:time                     _value:time  
        end_to_end_test.go:249: ------------------------------  ------------------------------  ----------------------  ----------------------  ------------------------------  ------------------------------  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k3                       m  2018-05-22T19:53:30.000000000Z  1970-01-01T00:00:00.000000000Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k3                       m  2018-05-22T19:53:31.000000000Z  1969-12-31T23:59:59.999999999Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k3                       m  2018-05-22T19:53:32.000000000Z  2018-05-22T19:53:26.033066000Z  
        end_to_end_test.go:249: 2018-05-22T19:52:00.000000000Z  2030-01-01T00:00:00.000000000Z                      k3                       m  2018-05-22T19:53:33.000000000Z  2018-05-22T18:00:00.000000000Z  
        end_to_end_test.go:252: table  got  not present in results
TestFluxEndToEnd - github.com/influxdata/influxdb/v2/query/stdlib/testing
Failed
=== RUN   TestFluxEndToEnd
ts=2020-09-16T22:50:14.750316Z lvl=info msg="Welcome to InfluxDB" log_id=0PIHcVNW000 version= commit= build_date=
ts=2020-09-16T22:50:14.751605Z lvl=info msg="Resources opened" log_id=0PIHcVNW000 service=bolt path=/tmp/660216617/influxd.bolt
ts=2020-09-16T22:50:14.751913Z lvl=info msg="Migration \"initial migration\" started (up)" log_id=0PIHcVNW000 service=migrations
ts=2020-09-16T22:50:14.752413Z lvl=info msg="Migration \"initial migration\" completed (up)" log_id=0PIHcVNW000 service=migrations
ts=2020-09-16T22:50:14.752468Z lvl=info msg="Migration \"add index \\\"userresourcemappingsbyuserindexv1\\\"\" started (up)" log_id=0PIHcVNW000 service=migrations
ts=2020-09-16T22:50:14.752761Z lvl=info msg="Migration \"add index \\\"userresourcemappingsbyuserindexv1\\\"\" completed (up)" log_id=0PIHcVNW000 service=migrations
ts=2020-09-16T22:50:14.752801Z lvl=info msg="Migration \"migrate task owner id\" started (up)" log_id=0PIHcVNW000 service=migrations
ts=2020-09-16T22:50:14.752980Z lvl=info msg="Migration \"migrate task owner id\" completed (up)" log_id=0PIHcVNW000 service=migrations
ts=2020-09-16T22:50:14.753015Z lvl=info msg="Migration \"create DBRP buckets\" started (up)" log_id=0PIHcVNW000 service=migrations
ts=2020-09-16T22:50:14.753106Z lvl=info msg="Migration \"create DBRP buckets\" completed (up)" log_id=0PIHcVNW000 service=migrations
ts=2020-09-16T22:50:14.753141Z lvl=info msg="Migration \"create pkger stacks buckets\" started (up)" log_id=0PIHcVNW000 service=migrations
ts=2020-09-16T22:50:14.753236Z lvl=info msg="Migration \"create pkger stacks buckets\" completed (up)" log_id=0PIHcVNW000 service=migrations
ts=2020-09-16T22:50:14.753269Z lvl=info msg="Migration \"delete sessionsv1 bucket\" started (up)" log_id=0PIHcVNW000 service=migrations
ts=2020-09-16T22:50:14.753350Z lvl=info msg="Migration \"delete sessionsv1 bucket\" completed (up)" log_id=0PIHcVNW000 service=migrations
ts=2020-09-16T22:50:14.753387Z lvl=info msg="Migration \"Create TSM metadata buckets\" started (up)" log_id=0PIHcVNW000 service=migrations
ts=2020-09-16T22:50:14.753470Z lvl=info msg="Migration \"Create TSM metadata buckets\" completed (up)" log_id=0PIHcVNW000 service=migrations
ts=2020-09-16T22:50:14.759146Z lvl=info msg="Using data dir" log_id=0PIHcVNW000 service=temporary_engine service=storage-engine path=/tmp/e2e386223220/data
ts=2020-09-16T22:50:14.759247Z lvl=info msg="Compaction settings" log_id=0PIHcVNW000 service=temporary_engine service=storage-engine max_concurrent_compactions=18 throughput_bytes_per_second=50331648 throughput_bytes_per_second_burst=50331648
ts=2020-09-16T22:50:14.759288Z lvl=info msg="Open store (start)" log_id=0PIHcVNW000 service=temporary_engine service=storage-engine op_name=tsdb_open op_event=start
ts=2020-09-16T22:50:14.759375Z lvl=info msg="Open store (end)" log_id=0PIHcVNW000 service=temporary_engine service=storage-engine op_name=tsdb_open op_event=end op_elapsed=0.092ms
ts=2020-09-16T22:50:14.759535Z lvl=info msg="Starting retention policy enforcement service" log_id=0PIHcVNW000 service=temporary_engine service=retention check_interval=30m
ts=2020-09-16T22:50:14.759661Z lvl=info msg="Starting precreation service" log_id=0PIHcVNW000 service=temporary_engine service=shard-precreation check_interval=10m advance_period=30m
ts=2020-09-16T22:50:14.759811Z lvl=info msg="Starting query controller" log_id=0PIHcVNW000 service=storage-reads concurrency_quota=10 initial_memory_bytes_quota_per_query=9223372036854775807 memory_bytes_quota_per_query=9223372036854775807 max_memory_bytes=0 queue_size=10
ts=2020-09-16T22:50:14.776764Z lvl=info msg="Configuring InfluxQL statement executor (zeros indicate unlimited)." log_id=0PIHcVNW000 max_select_point=0 max_select_series=0 max_select_buckets=0
ts=2020-09-16T22:50:15.080808Z lvl=info msg=Listening log_id=0PIHcVNW000 transport=http addr=127.0.0.1:0 port=42721
ts=2020-09-16T22:50:15.080925Z lvl=debug msg="user create" log_id=0PIHcVNW000 store=new took=0.180ms
ts=2020-09-16T22:50:16.019205Z lvl=debug msg="set password" log_id=0PIHcVNW000 store=new took=938.208ms
ts=2020-09-16T22:50:16.019555Z lvl=debug msg="org find by ID" log_id=0PIHcVNW000 store=new took=0.092ms
ts=2020-09-16T22:50:16.019794Z lvl=debug msg="bucket create" log_id=0PIHcVNW000 store=new took=0.333ms
ts=2020-09-16T22:50:16.020031Z lvl=debug msg="org find by ID" log_id=0PIHcVNW000 store=new took=0.076ms
ts=2020-09-16T22:50:16.020188Z lvl=debug msg="bucket create" log_id=0PIHcVNW000 store=new took=0.235ms
ts=2020-09-16T22:50:16.020484Z lvl=debug msg="urm create" log_id=0PIHcVNW000 store=new took=0.202ms
ts=2020-09-16T22:50:16.020542Z lvl=debug msg="org create" log_id=0PIHcVNW000 store=new took=1.256ms
ts=2020-09-16T22:50:16.020637Z lvl=debug msg="org find by ID" log_id=0PIHcVNW000 store=new took=0.055ms
ts=2020-09-16T22:50:16.020733Z lvl=debug msg="bucket create" log_id=0PIHcVNW000 store=new took=0.153ms
ts=2020-09-16T22:50:16.021371Z lvl=debug msg="onboard initial user" log_id=0PIHcVNW000 handler=onboard took=940.630ms
--- FAIL: TestFluxEndToEnd (114.62s)
```

--- a/query/stdlib/testing/end_to_end_test.go
+++ b/query/stdlib/testing/end_to_end_test.go
@@ -151,32 +151,47 @@ func makeTestPackage(file *ast.File) *ast.Package {
 	return pkg
 }
 
-var optionsSource = `
+// This options definition puts to() in the path of the CSV input. The tests
+// get run in this case and they would normally pass, if we checked the
+// results, but don't look at them.
+var writeOptSource = `
 import "testing"
 import c "csv"
-import "experimental"
-
-// Options bucket and org are defined dynamically per test
 
 option testing.loadStorage = (csv) => {
-	return experimental.chain(
-		first:  c.from(csv: csv) |> to(bucket: bucket, org: org),
-		second: from(bucket:bucket)
-	)
+	return c.from(csv: csv) |> to(bucket: bucket, org: org)
 }
 `
-var optionsAST *ast.File
 
-func init() {
+// This options definition is for the second run, the test run. It loads the
+// data from previously written bucket. We check the results after runnig this
+// second pass and report on them.
+var readOptSource = `
+import "testing"
+import c "csv"
+
+option testing.loadStorage = (csv) => {
+	return from(bucket: bucket)
+}
+`
+
+var writeOptAST *ast.File
+var readOptAST *ast.File
+
+func prepareOptions(optionsSource string) *ast.File {
 	pkg := parser.ParseSource(optionsSource)
 	if ast.Check(pkg) > 0 {
 		panic(ast.GetError(pkg))
 	}
-	optionsAST = pkg.Files[0]
+	return pkg.Files[0]
+}
+
+func init() {
+	writeOptAST = prepareOptions(writeOptSource)
+	readOptAST = prepareOptions(readOptSource)
 }
 
 func testFlux(t testing.TB, l *launcher.TestLauncher, file *ast.File) {
-
 	b := &platform.Bucket{
 		OrgID:           l.Org.ID,
 		Name:            t.Name(),
@@ -206,6 +221,34 @@ func testFlux(t testing.TB, l *launcher.TestLauncher, file *ast.File) {
 			Init: &ast.StringLiteral{Value: l.Org.Name},
 		},
 	}
+
+	executeWithOptions(t, l, bucketOpt, orgOpt, writeOptAST, file)
+
+	results := executeWithOptions(t, l, bucketOpt, orgOpt, readOptAST, file)
+	if results != nil {
+		logFormatted := func(name string, results map[string]*bytes.Buffer) {
+			if _, ok := results[name]; ok {
+				scanner := bufio.NewScanner(results[name])
+				for scanner.Scan() {
+					t.Log(scanner.Text())
+				}
+			} else {
+				t.Log("table ", name, " not present in results")
+			}
+		}
+		if _, ok := results["diff"]; ok {
+			t.Error("diff table was not empty")
+			logFormatted("diff", results)
+			logFormatted("want", results)
+			logFormatted("got", results)
+		}
+	}
+}
+
+func executeWithOptions(t testing.TB, l *launcher.TestLauncher, bucketOpt *ast.OptionStatement,
+	orgOpt *ast.OptionStatement, optionsAST *ast.File, file *ast.File) map[string]*bytes.Buffer {
+	var results map[string]*bytes.Buffer
+
 	options := optionsAST.Copy().(*ast.File)
 	options.Body = append([]ast.Statement{bucketOpt, orgOpt}, options.Body...)
 
@@ -230,7 +273,7 @@ func testFlux(t testing.TB, l *launcher.TestLauncher, file *ast.File) {
 	if r, err := l.FluxQueryService().Query(ctx, req); err != nil {
 		t.Fatal(err)
 	} else {
-		results := make(map[string]*bytes.Buffer)
+		results = make(map[string]*bytes.Buffer)
 
 		for r.More() {
 			v := r.Next()
@@ -246,22 +289,6 @@ func testFlux(t testing.TB, l *launcher.TestLauncher, file *ast.File) {
 		if err := r.Err(); err != nil {
 			t.Error(err)
 		}
-
-		logFormatted := func(name string, results map[string]*bytes.Buffer) {
-			if _, ok := results[name]; ok {
-				scanner := bufio.NewScanner(results[name])
-				for scanner.Scan() {
-					t.Log(scanner.Text())
-				}
-			} else {
-				t.Log("table ", name, " not present in results")
-			}
-		}
-		if _, ok := results["diff"]; ok {
-			t.Error("diff table was not empty")
-			logFormatted("diff", results)
-			logFormatted("want", results)
-			logFormatted("got", results)
-		}
 	}
+	return results
 }

--- a/query/stdlib/testing/testing.go
+++ b/query/stdlib/testing/testing.go
@@ -92,8 +92,6 @@ var FluxEndToEndSkipList = map[string]map[string]string{
 
 		"holt_winters_panic": "Expected output is an empty table which breaks the testing framework (https://github.com/influxdata/influxdb/issues/14749)",
 		"map_nulls":          "to cannot write null values",
-
-		"to_time": "Flaky test https://github.com/influxdata/influxdb/issues/19577",
 	},
 	"experimental": {
 		"set":       "Reason TBD",

